db/db_filesnapshot.cc

```diff
@@ -166,6 +166,12 @@ Status DBImpl::GetCurrentWalFile(std::unique_ptr<LogFile>* current_log_file) {
 Status DBImpl::GetLiveFilesStorageInfo(
     const LiveFilesStorageInfoOptions& opts,
     std::vector<LiveFileStorageInfo>* files) {
+  return GetLiveFilesStorageInfoHelper(opts, files, /* is_primary_db */ true);
```

Comment:
Can we get rid of the `is_primary_db` to make the logic simpler?
```
DBImpl::GetLiveFilesStorageInfo() {
  // Flush if necessary
  // Flush WAL if necessary
  return GetLiveFilesStorageInfoHelper();
}
```

Reply:
I believe this is legal (all writes completed before call to GetLiveFilesStorageInfo are persisted and included in checkpoint), but FlushWAL is ideally as late as possible to include as many writes as possible.

Also ideally, FlushWAL and FlushForGetLiveFiles would return NotSupported when not supported on the DB, and GetLiveFilesStorageInfo would allow proceeding in those cases.

If that's not practical at the moment, my preference is to call the extra parameter `is_flush_supported` referring both to memtable flush and WAL flush. Note that "flush" is not a logical write, so could (at least in my mind) be supported on a "read only" DB interface. But not a big deal.

With a test, I think is close to ready. Yanqin also mentioned we should check read-only DB support.